### PR TITLE
[ntsc-1.2] Match ntsc-1.2 BSS and add to CI

### PR DESCRIPTION
**Pull request by cadmic**
I shuffled some `g` and `s` names around based on evidence from bss ordering. And the size of `D_80858AD8` in z_player.c was wrong, apparently there's some padding after it in other versions but not in ntsc-1.2.

### CODE REVIEW

File: src/overlays/actors/ovl_player_actor/z_player.c [hunk 1]
```diff
@@ -14262,7 +14265,7 @@ void Player_Action_80850AEC(Player* this, PlayState* play) {
 
 void Player_Action_80850C68(Player* this, PlayState* play) {
     if ((this->av2.actionVar2 != 0) && ((this->unk_858 != 0.0f) || (this->unk_85C != 0.0f))) {
-        static Vec3s D_80858AD8[25];
+        static Vec3s D_80858AD8[24];
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_player_actor/z_player.c:14268]: sus change, this buffer is passed to `LinkAnimation_BlendToMorph` which then 16-aligns it and (`AnimTaskQueue_AddLoadPlayerFrame`) loads `sizeof(Vec3s)*skelAnime->limbCount+2` bytes into it

`skelAnime->limbCount` has value 22:
- `gLinkAdultSkel.sh.limbCount = 21`
- `SkelAnime_InitLink(flags=9)` sets `skelAnime->limbCount = 1 + gLinkAdultSkel.sh.limbCount`

so, after 16-alignment, there much be `6*22+2=134` bytes available

before alignment, this buffer is `24*6=144` bytes. There's only 10 bytes of margin, not enough to even cover all cases if 4-aligned (for example if aligned to 4 modulo 16, 12 bytes would be discarded, leaving 132 bytes in the buffer, but 134 will be used)

And if IDO 4-aligns stuff, I think gcc would 2-align this (Vec3s is made of s16s)

Comment (cadmic) [reply src/overlays/actors/ovl_player_actor/z_player.c:14268]: Well there does seem to be only 144 bytes between this symbol and the next one in ntsc-1.2:
```
Baserom BSS ordering:
  offset=0x0000 size=0x0004 align=0x4 referenced_in_data=False D_80858AA4
  offset=0x0008 size=0x000C align=0x8 referenced_in_data=False sInteractWallCheckResult
  offset=0x0014 size=0x0004 align=0x4 referenced_in_data=False sControlInput
  offset=0x0018 size=0x0004 align=0x4 referenced_in_data=False D_80858AA0
  offset=0x0020 size=0x0090 align=0x8 referenced_in_data=False Player_Action_80850C68::D_80858AD8
  offset=0x00B0 size=0x000C align=0x8 referenced_in_data=False Player_Update::sDogSpawnPos
  offset=0x00C0 size=0x000C align=0x8 referenced_in_data=False sBunnyEarKinematics
```
Maybe it should be `u64 D_80858AD8[18]` so both IDO and GCC align this "safely"?

Comment (Dragorn421) [reply src/overlays/actors/ovl_player_actor/z_player.c:14268]: `u64 D_80858AD8[18]` sounds good

**Review by fig02 — APPROVED** [review]
fully matching after this?

**Comment by cadmic** [comment]
Yep, I removed the `COMPARE := 0` line in the Makefile so the hash is now checked.

I noticed that a few files seemed to have fragile bss in ntsc-1.2, so I tried to reorganize things a little bit to hopefully make it more stable. I couldn't do much for z_camera or z_std_dma though unfortunately

File: src/code/main.c [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 
-#pragma increment_block_number "gc-eu:80 gc-eu-mq:80 gc-jp:80 gc-jp-ce:80 gc-jp-mq:80 gc-us:80 gc-us-mq:80"
+// Declared before including other headers for BSS ordering
+extern uintptr_t gSegments[NUM_SEGMENTS];
 
-struct PreNmiBuff* gAppNmiBufferPtr;
+#pragma increment_block_number "gc-eu:252 gc-eu-mq:252 gc-jp:252 gc-jp-ce:252 gc-jp-mq:252 gc-us:252 gc-us-mq:252" \
+                               "ntsc-1.2:128"
 
-#include "segmented_address.h"
```
[the remark this answers is not in the document]
Comment (cadmic) [reply src/code/main.c:7]: oh, guess I didn't notice this when merging (why was this not with the other includes?)

[the remark this answers is not in the document]
Comment (cadmic) [reply src/code/main.c:7]: nvm, I assume it was for bss ordering

**Review by Dragorn421 — APPROVED** [review]

**Pull request merged**